### PR TITLE
add kancolle api

### PR DESCRIPTION
**Pull request by myangelkamikaze**
- renamed a few types cause swagger doesn't let you have the same name multiple times even if it's valid for different namespaces
- added explicit `<inheritdoc />` cause swagger can't get the comments without it
- the KancolleApi project is basically just to generate the swagger doc for the [KancolleApi repo](https://github.com/ElectronicObserverEN/KancolleApi)

### CODE REVIEW

File: ElectronicObserver.KancolleApi.Types/ApiReqMap/Next/APIReqMapNextResponse.cs [hunk 1]
```diff
@@ -34,9 +35,11 @@ public class ApiReqMapNextResponse : IMapProgressApi
 	[JsonPropertyName("api_destruction_battle")]
 	public ApiDestructionBattle? ApiDestructionBattle { get; set; }
 
+	/// <inheritdoc />
 	[JsonPropertyName("api_event_id")]
 	public int ApiEventId { get; set; }
 
+	/// <inheritdoc />apihap
```
Comment (Jebzou) [inline ElectronicObserver.KancolleApi.Types/ApiReqMap/Next/APIReqMapNextResponse.cs:42]: apihap ?

**Comment by Jebzou** [comment]
I get an error when running the API project "**API Explorer not registered in DI**"
Adding 
```cs 
builder.Services.AddEndpointsApiExplorer();
``` 
seems to fix it ?

File: ElectronicObserver.KancolleApi.Types/ApiReqMap/Next/APIReqMapNextResponse.cs (hunk 1 above)
Comment (myangelkamikaze) [reply ElectronicObserver.KancolleApi.Types/ApiReqMap/Next/APIReqMapNextResponse.cs]: huh, wonder how that happened

**Comment by myangelkamikaze** [comment]
> I get an error when running the API project "**API Explorer not registered in DI**" Adding
> 
> ```cs
> builder.Services.AddEndpointsApiExplorer();
> ```
> 
> seems to fix it ?

Hmm, I couldn't repro the error but read online that it's supposedly needed?
No idea why it worked on my side without that.

**Review by Jebzou — APPROVED** [review]

**Pull request merged**